modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/codec/AmqpChannel.java

```diff
@@ -410,6 +410,10 @@ public void setDistributedTransactional() {
      */
     public void commit() throws ValidationException, BrokerException {
         transaction.commit();
+        updatePendingAcknowledgements();
+    }
+
+    private void updatePendingAcknowledgements() {
```

Comment:
Maybe this should be renamed as `releasePendingAcknowledgements()`.

Reply:
Fixed!